drivers/samsung/displays/mdc_protocol.cr

```diff
@@ -0,0 +1,376 @@
+    Widi          = 0x61
+    Magic_Info    = 0x20
+    Whiteboard    = 0x64
+  end
+  include PlaceOS::Driver::Interface::InputSelection(Inputs)
+
+  # Discovery Information
+  tcp_port 1515
+  descriptive_name "Samsung MD, DM & QM Series LCD"
+  generic_name :Display
+
+  # Markdown description
+  description <<-DESC
+  For DM displays configure the following 1:
+
+  1. Network Standby = ON
+  2. Set Auto Standby = OFF
+  3. Set Eco Solution, Auto Off = OFF
+
+  Hard Power off displays each night and hard power ON in the morning.
+  DESC
+
+  default_settings({
+    display_id: 0,
+    rs232_control: false
+  })
+
+  @id : Int32 = 0
+  @rs232 : Bool = false
+  @blank : Input?
+  @previous_volume : Int32 = 50
+  # Meta data for inquiring interfaces
+  @input_stable : Bool = true
+  @input_target : Input = Input::Hdmi
+  @power_stable : Bool = true
+
+  def on_load
+    transport.tokenizer = Tokenizer.new do |io|
+      bytes = io.peek
+      # Ensure message indicator is well-formed
+      disconnect unless bytes.first == INDICATOR
+      logger.debug { "Received: #{bytes}" }
+      # [header, command, id, data.size, [data], checksum]
+      # return 0 if the message is incomplete
+      bytes.size < 4 ? 0 : bytes[3].to_i + 5
+    end
+
+    on_update
+  end
+
+  def on_update
+    @id = setting(Int32, :display_id)
+    @rs232 = setting(Bool, :rs232_control)
+    if blanking_input = setting?(String, :blanking_input)
+      @blank = Input.parse?(blanking_input)
+    end
+  end
+
+  def connected
+    do_device_config unless self[:hard_off]?
+
+    schedule.every(30.seconds, true) do
+      do_poll
+    end
+  end
+
+  def disconnected
+    self[:power] = false unless @rs232
+    schedule.clear
+  end
+
+  # As true power off disconnects the server we only want to
+  # power off the panel. This doesn't work in video walls
+  # so if a nominal blank input is
+  # TODO: check type for broadcast is correct
+  def power(power : Bool, broadcast : String? = nil)
+    self[:power_target] = power
+    @power_stable = false
+
+    if !power
+      # Blank the screen before turning off panel if required
+      # required by some video walls where screens are chained
+      if (blanking_input = @blank) && self[:power]?
+        switch_to(blanking_input)
+      end
+      do_send(Command::Panel_Mute, 1)
+    elsif !@rs232 && !self[:connected]? && (broadcast_string = broadcast)
+       wake_device(broadcast_string)
+    else
+      # Power on
+      do_send(Command::Hard_Off, 1)
+      do_send(Command::Panel_Mute, 0)
+    end
+  end
+
+  def hard_off
+    do_send(Command::Panel_Mute, 0) if self[:power]?
+    do_send(Command::Hard_Off, 0)
+  end
+
+  def power?(**options)
+    do_send(Command::Panel_Mute, Bytes.empty, **options).get
+    self[:power]
+  end
+
+  # Mutes audio + video
+  def mute(
+    state : Bool = true,
+    index : Int32 | String = 0,
+    layer : MuteLayer = MuteLayer::AudioVideo
+  )
+    logger.debug { "requested mute state: #{state}" }
```

Comment:
For consistency these debug messages should exist across all device interactions, or none.